test/injector/should-include-component-based/input.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+
+// it's technically not correct since this descripts props the component
+// sees not just the one available to the user. We're abusing this to provide
+// some concrete documentation for `key` regarding this component
+export interface SnackBarProps {
+	/**
+	 * some hints about state reset that relates to prop of this component
+	 */
+	key?: any;
+	/**
+	 * some prop that is inherited which we don't care about here
+	 */
+	onChange?: () => any;
+}
+
+export function Snackbar(props: SnackBarProps) {
+	return <div {...props} />;
+}
+
+// here we don't care about `key`
+export function SomeOtherComponent(props: { children?: React.ReactNode }) {
```

Comment:
It shows how we care about `key` in one component but not the other. Without `component` we wouldn't be able to differentiate (unless we would match the jsdoc which is not robust).

Reply:
Yeah, but the component doesn't have a `key` prop. Which is why I think you meant to have it use `SnackBarProps`

Reply:
Do you infer the props that can be passed to `createElement` or short-circuit type inference and go directly to the props interface? Because every component has a `key` prop implicitly.

Reply:
It only parses declared types, so as you can see from `output.json`, `SomeOtherComponent` doesn't have a prop named `key`

Reply:
Well then let's stick with children. Either way this should not use SnackBarProps.

Reply:
done